### PR TITLE
Feature/multi id vanes

### PR DESCRIPTION
**Pull request by acfloria**
- Add starting up the hall effect sensor in rc.sensors
- Extend the hall driver to accept different i2c addresses
- Implement handling the hall measurements in the sensors module and publish the airflow angles messages
- Add the airflow angles mavlink stream

Tested in lab with two different vanes with EZG3

### CODE REVIEW

File: src/modules/sensors/sensors.cpp [hunk 1]
```diff
@@ -368,13 +480,13 @@ void Sensors::diff_pres_poll()
...
 		/* don't risk to feed negative airspeed into the system */
 		airspeed.indicated_airspeed_m_s = calc_IAS_corrected((enum AIRSPEED_COMPENSATION_MODEL)
-						  _parameters.air_cmodel,
-						  smodel, _parameters.air_tube_length, _parameters.air_tube_diameter_mm,
-						  diff_pres.differential_pressure_filtered_pa, air_data.baro_pressure_pa,
-						  air_temperature_celsius);
+                                                 _parameters.air_cmodel,
+                                                 smodel, _parameters.air_tube_length, _parameters.air_tube_diameter_mm,
+                                                 diff_pres.differential_pressure_filtered_pa, air_data.baro_pressure_pa,
+                                                 air_temperature_celsius);
```
Comment (Jaeyoung-Lim) [inline src/modules/sensors/sensors.cpp:486]: Unnecessary whitespace changes?

File: src/modules/sensors/sensors.cpp [hunk 2]
```diff
@@ -330,7 +442,7 @@ void Sensors::diff_pres_poll()
 		} else {
 			// differential pressure temperature invalid, check barometer
 			if ((air_data.timestamp != 0) && PX4_ISFINITE(air_data.baro_temp_celcius)
-			    && (air_data.baro_temp_celcius >= -40.f) && (air_data.baro_temp_celcius <= 125.f)) {
+                           && (air_data.baro_temp_celcius >= -40.f) && (air_data.baro_temp_celcius <= 125.f)) {
```
Comment (Jaeyoung-Lim) [inline src/modules/sensors/sensors.cpp:445]: Unnecessary whitespace changes?

File: msg/airflow_slip.msg [hunk 3]
```diff
@@ -0,0 +1,3 @@
+uint64 timestamp		# time since system start (microseconds)
+float32 slip_rad # sideslip angle in radians
+bool valid # true if measurement is within sensor/calibration range, false otherwise
```
Comment (Jaeyoung-Lim) [inline msg/airflow_slip.msg:3]: If the `airflow_slip` and `airflow_aoa` have identical data/information, can we share a single uORB message?

File: src/modules/sensors/sensors.cpp [hunk 4]
```diff
@@ -156,6 +170,26 @@ class Sensors : public ModuleBase<Sensors>, public ModuleParams, public px4::Sch
...
+
+		int32_t CAL_AV_SLIP_ID;
+		float CAL_AV_SLIP_OFF;
+		float CAL_AV_SLIP_MIN;
+		float CAL_AV_SLIP_MAX;
+		float CAL_AV_SLIP_REV;
+		float CAL_AV_SLIP_P0;
+		float CAL_AV_SLIP_P1;
+		float CAL_AV_SLIP_P2;
+		float CAL_AV_SLIP_P3;
```
Comment (Jaeyoung-Lim) [inline src/modules/sensors/sensors.cpp:192]: Would this fit better in a struct?

File: src/modules/mavlink/streams/SENSOR_AIRFLOW_ANGLES.hpp [hunk 5]
```diff
@@ -0,0 +1,105 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```
Comment (Jaeyoung-Lim) [inline src/modules/mavlink/streams/SENSOR_AIRFLOW_ANGLES.hpp:3]:
```suggestion
 *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```

**Review by Jaeyoung-Lim — COMMENTED** [review]
Thanks!

Just some minor review comments.
- I think AOA and Slip sensors are basically the same sensor/data, but they are treated differently in this PR. IMHO since it is actually the same hardware, they should be identical but handled differently only from the estimator side. Therefore I think it would be better if we can share the uORB message definition, and if possible calibration parameters whether the sensor is defined as a sideslip sensor or a aoa sensor.

- Format checks are failing and therefore the tests are not running. It would be nice if we can see the test results

File: src/modules/sensors/sensors.cpp (hunk 4 above)
Comment (acfloria) [reply src/modules/sensors/sensors.cpp:192]: It is already part of the Parameters strucht, right? I don't think making a struct in a struct makes sense.

**Comment by acfloria** [comment]
I updated the formatting with astyle.

I would not be totally opposed to changing the airflow angles to the same ID and adding an additional field indicating if it is the aoa or slip. However, for the moment I would like to keep it as is, since the whole post processing pipeline works with those two uOrb message types and we would have to change it quite a bit to work with this new definition. We could do this change in a new PR after everything is tested in the current form, which is the state on our older firmware.

At some point we will have to associate the measurements either with the aoa or slip and I think the sensors module is a good way to do it.

**Review by Jaeyoung-Lim — CHANGES_REQUESTED** [review]
Would it be possible to address the build failures?

The build is failing because the new uORB messages are not aaded to the rtps message definitions
```
AssertionError: 
The following messages are not listen under  /__w/ethzasl_fw_px4/ethzasl_fw_px4/msg/tools/uorb_rtps_message_ids.yaml: airflow_slip, airflow_aoa

Please add them to the yaml file with the respective ID and, if applicable, mark them to be sent or received by the micro-RTPS bridge.
NOTE: If the message is an alias / part of multi-topics (#TOPICS) of a base message, it should be added as well.

```

**Comment by Jaeyoung-Lim** [comment]
Merging to address addiitonal changes in https://github.com/ethz-asl/ethzasl_fw_px4/pull/18
